onnxscript/ir/_linked_list.py

```diff
@@ -136,27 +136,12 @@ def __len__(self) -> int:
         )
         return self._length
 
-    def __getitem__(self, index: int) -> T:
+    def __getitem__(self, index: int | slice) -> T:
```

Comment:
Could you create overloads so the type annotations is correct? Two overloads:

`def __getitem__(self, index: int) -> T:`

and `def __getitem__(self, index: slice) -> tuple[T, ...]:`